inference/core/workflows/core_steps/flow_control/rate_limiter/v1.py

```diff
@@ -99,8 +109,18 @@ def run(
         cooldown_seconds: float,
         depends_on: any,
         next_steps: List[StepSelector],
+        video_reference_image: Optional[WorkflowImageData] = None,
     ) -> BlockResult:
         current_time = datetime.now()
+        try:
+            metadata = video_reference_image.video_metadata
```

Comment:
should we add a check/if for the optional value here rather than relying on exception handling.  Not sure about python internals, but could see this being slower / less efficient than explicit check?

Reply:
I wasn't sure, so I asked ChatGPT:
> In Python, there's a well-known coding style preference known as "Easier to Ask for Forgiveness than Permission" (EAFP), which encourages using try/except blocks to handle unexpected states rather than checking conditions ahead of time. This style is often considered more idiomatic and can make code more concise.

I had it benchmark it. Over 1 million iterations, it was a tiny bit faster when metadata was present & a bit slower (~1.6 microseconds per frame) when metadata was not present. Given this should likely be set most of the time (rate limiter really makes most sense when running on video) I think this should be fine.